src/main/java/com/celements/webform/DocFormScriptService.java

```diff
@@ -90,4 +126,42 @@ private DocFormCommand getDocFormCommand() {
     }
     return (DocFormCommand) getContext().get(_DOC_FORM_COMMAND_OBJECT);
   }
+
+  Map<String, Set<DocumentReference>> saveXWikiDocCollection(
```

Comment:
Please split saveXWikiDocCollection in two methods and move them to appropriate class:
1. hasEditOnAllDocs -> belongs into ScriptService 
2. saveXWikiDocCollection -> only saving the docs without rights checks belongs to DocFormCommand class. It must be possible to access saveXWikiDocCollection out of java without rights checks.